### ISSUE TITLE
java.lang.NoClassDefFoundError: liquibase/Scope Liquibase 3.5.2 Spring Boot 1.5.15

### ISSUE THREAD

**Issue by eninja** [blagerweij/liquibase-sessionlock#19]
Thanks for this extension! 
Perhaps more a question than a bug: is there a minimum version of liquibase it is compatible with?

I have no problem with:
Spring Boot 2.7.0
Liquibase 4.9.1

but when I added it to:
Spring Boot 1.5.15
Liquibase 3.5.2

Application run failed with exception: 
```
2022-08-11 15:05:38.601 ERROR [document-service,,,] 7 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is java.lang.NoClassDefFoundError: liquibase/Scope
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1778)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:593)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:515)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:320)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:318)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:307)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1105)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:867)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:549)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:140)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:742)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:389)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:311)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1213)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1202)
	at pl.com.euro.kl2.document.DocumentService.main(DocumentService.java:16)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:47)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:86)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: java.lang.NoClassDefFoundError: liquibase/Scope
	at com.github.blagerweij.sessionlock.SessionLockService.getLog(SessionLockService.java:157)
	at com.github.blagerweij.sessionlock.SessionLockService.acquireLock(SessionLockService.java:81)
	at liquibase.lockservice.StandardLockService.waitForLock(StandardLockService.java:207)
	at liquibase.Liquibase.update(Liquibase.java:184)
	at liquibase.Liquibase.update(Liquibase.java:179)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:353)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:305)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1837)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1774)
	... 26 common frames omitted
Caused by: java.lang.ClassNotFoundException: liquibase.Scope
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:92)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 35 common frames omitted
```

I have also tried add system property liquibase.scan.packages, but result is the same.

**Comment by jmitash** [comment]
The issue is that this repository depends on the `liquibase.Scope` class, but that class was only added in Liquibase 3.7.

In other words, in its current state, Liquibase Sessionlock is not compatible with Liquibase 3.6 or below.

**Comment by eninja** [comment]
Thanks, 
I have created https://github.com/blagerweij/liquibase-sessionlock/pull/20 to clarify this

**Comment by blagerweij** [comment]
Hey @eninja , thanks for the report. I've added some reflection magic to detect older Liquibase versions, so that loading the extension does not break on older versions. Also, I've added a few older versions to the build matrix to verify compilation and unit-tests. Please let me know whether the sessionlock works on 3.5.x.

**Comment by blagerweij** [comment]
Fixed by #22

**Comment by eninja** [comment]
@blagerweij thank you very much! it works with dependencies mentioned above.
